spanner/api/Spanner.Samples.Tests/GetDatabaseDdlAsyncTest.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Threading.Tasks;
+using Xunit;
+
+[Collection(nameof(SpannerFixture))]
+public class GetDatabaseDdlAsyncTest
+{
+    private readonly SpannerFixture _spannerFixture;
+
+    public GetDatabaseDdlAsyncTest(SpannerFixture spannerFixture)
+    {
+        _spannerFixture = spannerFixture;
+    }
+
+    [Fact]
+    public async Task TestGetDatabaseDdlAsync()
+    {
+        var createSingersTable =
+            @"CREATE TABLE Singers (
+                     SingerId INT64 NOT NULL,
+                     FirstName STRING(1024),
+                     LastName STRING(1024),
+                     ComposerInfo BYTES(MAX),
+                 ) PRIMARY KEY (SingerId)";
+        var createAlbumsTable =
+            @"CREATE TABLE Albums (
+                     SingerId INT64 NOT NULL,
+                     AlbumId INT64 NOT NULL,
+                     AlbumTitle STRING(MAX),
+                 ) PRIMARY KEY (SingerId, AlbumId),
+                 INTERLEAVE IN PARENT Singers ON DELETE CASCADE";
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(_spannerFixture.InstanceId, async databaseId =>
```

Comment:
Note: This creates a temporary database, executes the test function and then deletes the test database. That might seem inefficient, but creating and dropping (empty) databases is relatively quick on Cloud Spanner nowadays. The execution time for only this test when measured from my local development machine is between 5-7 seconds excluding the time spent setting up the `SpannerFixture`. The total execution time for the test, including the initialisation that is done in `SpannerFixture`, is however around 4 minutes when it is executed multiple times (that is; the tables, backups, etc. have already been created by previous runs).
For this type of test that does not depend on any large data model or previously inserted data, it would be good to have a possibility of skipping most of the initialisation in the `SpannerFixture` in order to speed up the development of new samples and tests.

Reply:
Hmm, somehow I missed this comment when I reviewed earlier.

I think the best option we have here is to have different fixtures classes, a base one, for the common code and initialization (Project Id, for instance, etc..) and then specific ones for the biggest use cases. This is what we tend to do in libraries.
So maybe we have a fixture that initializes the backups, and maybe another that initializes the multi region instance, etc...

Any of that of course can be done in a separate PR, but I agree that our current fixture is getting a little bit overwhelming in terms of how long it takes it to initialize and how much (unrelated) code it is holding.

Reply:
@olavloite I think most of the outstanding comments are related to this suggestion here of splitting the fixture into more logical units and better managing initialization and deletion there. I'm happy to merge this PR as is if we can work on this after. Else, I'll probably take a closer look at a couple of things (nothing major, just want to make sure that we are consistent in our approach to resources, etc.)

Reply:
I'm certainly open to working on splitting the fixture into more logical units in one or more separate PRs. Your suggestion to split it into a base for ProjectId etc., and a couple of more feature specific fixtures sounds good. Just let me know if you want me to start with a draft PR, or if you want to start with a suggestion.

Reply:
Perfect. Before starting any work, I'm creating an issue so we can track our preferences there and reach an agreement beforehand. Thanks!